libtransact/src/scheduler/serial/execution.rs

```diff
@@ -80,7 +80,7 @@ impl ExecutionTaskCompletionNotifier for SerialExecutionTaskCompletionNotifier {
     fn notify(&self, notification: ExecutionTaskCompletionNotification) {
         self.tx
             .send(CoreMessage::ExecutionResult(notification))
-            .unwrap();
+            .unwrap_or_else(|err| error!("failed to send notification to core: {}", err));
```

Comment:
The notifier and the task iterator should probably have a way of reporting errors as well; this could probably be done through the error callback mechanism, but the mechanics of that may be tricky.